app/scripts/platforms/extension.js

```diff
@@ -13,70 +13,52 @@ export default class ExtensionPlatform {
     browser.runtime.reload();
   }
 
-  openTab(options) {
-    return new Promise((resolve, reject) => {
-      browser.tabs.create(options).then((newTab) => {
-        const error = checkForError();
-        if (error) {
-          return reject(error);
-        }
-        return resolve(newTab);
-      });
-    });
+  async openTab(options) {
+    const newTab = await browser.tabs.create(options);
+    const error = checkForError();
```

Comment:
I don't think we need the `checkForError` steps anymore when using the Promise-based APIs.
See: https://github.com/mozilla/webextension-polyfill#using-the-promise-based-apis

Reply:
Nice!